examples/06-fetch-map/legend.ts

```diff
@@ -1,107 +1,106 @@
-      tilestats &&
-      tilestats.type === 'Number' &&
-      tilestats.min !== undefined
-    ) {
-      const numRanges = 6;
-      const min = tilestats.min;
-      const max = tilestats.max;
-      const step = (max - min) / numRanges;
-
-      for (let i = 0; i < numRanges; i++) {
-        const rangeStart = min + step * i;
-        const rangeEnd = i === numRanges - 1 ? max : min + step * (i + 1);
-
-        const rangeDiv = document.createElement('div');
-        rangeDiv.className = 'legend-range';
-
-        const colorSwatch = document.createElement('div');
-        colorSwatch.className = 'legend-color-swatch';
-
-        const value = (rangeStart + rangeEnd) / 2;
-        const color = layer.props.getFillColor({
-          properties: {[dataColumn]: value},
-        });
-        colorSwatch.style.backgroundColor = `rgb(${color[0]}, ${color[1]}, ${color[2]})`;
-
-        const rangeLabel = document.createElement('div');
-        rangeLabel.className = 'legend-range-label';
-        rangeLabel.textContent = `${rangeStart.toFixed(2)} – ${rangeEnd.toFixed(2)}`;
-
-        rangeDiv.appendChild(colorSwatch);
-        rangeDiv.appendChild(rangeLabel);
-        layerDiv.appendChild(rangeDiv);
+    const scales = layer.scales || {};
+    for (const scaleKey of Object.keys(scales)) {
+      const scaleInfo = scales[scaleKey];
+      const dataColumn = scaleInfo?.field?.name;
+      const layerDiv = createLegendHeader(layer, dataColumn);
+
+      if (!dataColumn) {
+        const color = layer.props.getFillColor;
+        const rgb = `rgb(${color[0]}, ${color[1]}, ${color[2]})`;
+        layerDiv.appendChild(createColorSwatch(rgb, layer.props.cartoLabel));
+      } else {
+        const {domain, range, type} = scaleInfo;
+        if (
+          type === 'ordinal' ||
+          type === 'point' ||
+          typeof domain[0] === 'string'
+        ) {
+          // Simple one to one mapping
+          for (let i = 0; i < domain.length; i++) {
+            layerDiv.appendChild(createColorSwatch(range[i], domain[i]));
+          }
+        } else if (type === 'custom' || type === 'quantile') {
+          // Custom threshold scale: domain is [start1, start2, ..., null], range is colors
+          for (let i = 0; i < range.length; i++) {
+            let start: number;
+            let end: number;
+            if (type === 'custom') {
+              start = domain[i - 1] || 0;
+              end = domain[i] || Infinity;
+            } else {
+              start = domain[i];
+              end = domain[i + 1];
+            }
+            const labelText =
+              end === null || end === undefined || end === Infinity
+                ? `>${start.toFixed(1)}`
+                : `${start.toFixed(1)} – ${end.toFixed(1)}`;
+            layerDiv.appendChild(createColorSwatch(range[i], labelText));
+          }
+        } else if (
+          typeof domain[0] === 'number' &&
+          typeof domain[1] === 'number' &&
+          range.length > 1
```

Comment:
Similarly, just wondering if `type === 'linear'` would have the same effect here?

Reply:
Done, I also added `quantize`